Binary_sv2: std fails to compile

```rust
Compiling binary_sv2 v5.0.1 (/home/shourya/stratum/sv2/binary-sv2)
error[E0369]: binary operation `==` cannot be applied to type `std::io::Error`
   --> sv2/binary-sv2/src/lib.rs:253:13
    |
202 | #[derive(Debug, PartialEq, Eq, Clone)]
    |                 --------- in this derive macro expansion
...
253 |     IoError(E),
    |             ^
    |
    = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::io::Error: std::cmp::Eq` is not satisfied
   --> sv2/binary-sv2/src/lib.rs:253:13
    |
202 | #[derive(Debug, PartialEq, Eq, Clone)]
    |                            -- in this derive macro expansion
...
253 |     IoError(E),
    |             ^ the trait `std::cmp::Eq` is not implemented for `std::io::Error`
    |
note: required by a bound in `AssertParamIsEq`
   --> /home/shourya/.rustup/toolchains/1.75.0-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:336:31
    |
336 | pub struct AssertParamIsEq<T: Eq + ?Sized> {
    |                               ^^ required by this bound in `AssertParamIsEq`
    = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::io::Error: Clone` is not satisfied
   --> sv2/binary-sv2/src/lib.rs:253:13
    |
202 | #[derive(Debug, PartialEq, Eq, Clone)]
    |                                ----- in this derive macro expansion
...
253 |     IoError(E),
    |             ^ the trait `Clone` is not implemented for `std::io::Error`
    |
    = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0277, E0369.
For more information about an error, try `rustc --explain E0277`.
```

--- a/stratum-core/Cargo.toml
+++ b/stratum-core/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "stratum-core"
-version = "0.4.0"
+version = "0.5.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -13,20 +13,20 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 
 [dependencies]
 buffer_sv2 = { path = "../sv2/buffer-sv2", version = "^3.0.0" }
-binary_sv2 = { path = "../sv2/binary-sv2", version = "^5.0.0" }
-codec_sv2 = { path = "../sv2/codec-sv2", version = "^5.0.0", features = ["noise_sv2"]}
-extensions_sv2 = { path = "../sv2/extensions-sv2", version = "^0.1.0" }
-framing_sv2 = { path = "../sv2/framing-sv2", version = "^6.0.0" }
+binary_sv2 = { path = "../sv2/binary-sv2", version = "^6.0.0" }
+codec_sv2 = { path = "../sv2/codec-sv2", version = "^6.0.0", features = ["noise_sv2"]}
+extensions_sv2 = { path = "../sv2/extensions-sv2", version = "^0.2.0" }
+framing_sv2 = { path = "../sv2/framing-sv2", version = "^7.0.0" }
 noise_sv2 = { path = "../sv2/noise-sv2", version = "^1.0.0" }
-parsers_sv2 = { path = "../sv2/parsers-sv2", version = "^0.4.0" }
-handlers_sv2 = { path = "../sv2/handlers-sv2", version = "^0.4.0" }
-channels_sv2 = { path = "../sv2/channels-sv2", version = "^6.0.0" }
-common_messages_sv2 = { path = "../sv2/subprotocols/common-messages", version = "^7.2.0" }
-mining_sv2 = { path = "../sv2/subprotocols/mining", version = "^10.0.0" }
-template_distribution_sv2 = { path = "../sv2/subprotocols/template-distribution", version = "^5.1.0" }
-job_declaration_sv2 = { path = "../sv2/subprotocols/job-declaration", version = "^8.0.0" }
-sv1_api = { path = "../sv1", version = "^4.0.0", optional = true }
-stratum_translation = { path = "stratum-translation", version = "^0.3.0", optional = true }
+parsers_sv2 = { path = "../sv2/parsers-sv2", version = "^0.5.0" }
+handlers_sv2 = { path = "../sv2/handlers-sv2", version = "^0.5.0" }
+channels_sv2 = { path = "../sv2/channels-sv2", version = "^7.0.0" }
+common_messages_sv2 = { path = "../sv2/subprotocols/common-messages", version = "^8.0.0" }
+mining_sv2 = { path = "../sv2/subprotocols/mining", version = "^11.0.0" }
+template_distribution_sv2 = { path = "../sv2/subprotocols/template-distribution", version = "^6.0.0" }
+job_declaration_sv2 = { path = "../sv2/subprotocols/job-declaration", version = "^9.0.0" }
+sv1_api = { path = "../sv1", version = "^5.0.0", optional = true }
+stratum_translation = { path = "stratum-translation", version = "^0.4.0", optional = true }
 bitcoin = { workspace = true }
 
 [features]

--- a/stratum-core/stratum-translation/Cargo.toml
+++ b/stratum-core/stratum-translation/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "stratum_translation"
-version = "0.3.0"
+version = "0.4.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 description = "Stratum V1 ↔ Stratum V2 translation utilities for reuse across proxies, apps, and firmware"
@@ -10,9 +10,9 @@ name = "stratum_translation"
 path = "src/lib.rs"
 
 [dependencies]
-binary_sv2 = { path = "../../sv2/binary-sv2", version = "^5.0.0" }
-mining_sv2 = { path = "../../sv2/subprotocols/mining", version = "^10.0.0" }
-channels_sv2 = { path = "../../sv2/channels-sv2", version = "^6.0.0" }
-v1 = { path = "../../sv1", package = "sv1_api", version = "^4.0.0" }
+binary_sv2 = { path = "../../sv2/binary-sv2", version = "^6.0.0" }
+mining_sv2 = { path = "../../sv2/subprotocols/mining", version = "^11.0.0" }
+channels_sv2 = { path = "../../sv2/channels-sv2", version = "^7.0.0" }
+v1 = { path = "../../sv1", package = "sv1_api", version = "^5.0.0" }
 tracing = { workspace = true }
 bitcoin = { workspace = true }

--- a/sv1/Cargo.toml
+++ b/sv1/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "sv1_api"
-version = "4.0.0"
+version = "5.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -14,7 +14,7 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-binary_sv2 = { path = "../sv2/binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../sv2/binary-sv2", version = "^6.0.0" }
 bitcoin_hashes = { workspace = true }
 byteorder = { workspace = true }
 serde = { workspace = true }

--- a/sv1/src/methods/mod.rs
+++ b/sv1/src/methods/mod.rs
@@ -8,7 +8,7 @@ pub mod server_to_client;
 use crate::json_rpc::{Message, Response};
 
 /// Errors encountered during conversion between valid json_rpc messages and Sv1 messages.
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub enum MethodError<'a> {
     /// If the json_rpc message call a method not defined by Sv1. It contains the called method
     MethodNotFound(String),
@@ -35,7 +35,7 @@ impl From<binary_sv2::Error> for ParsingMethodError {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub enum ParsingMethodError {
     BadU256Convert(Box<binary_sv2::Error>),
     HexError(Box<BTCHashError>),

--- a/sv2/binary-sv2/Cargo.toml
+++ b/sv2/binary-sv2/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "binary_sv2"
-version = "5.0.1"
+version = "6.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"

--- a/sv2/binary-sv2/src/lib.rs
+++ b/sv2/binary-sv2/src/lib.rs
@@ -83,6 +83,40 @@ pub use crate::codec::{
 
 use alloc::vec::Vec;
 
+#[cfg(not(feature = "no_std"))]
+#[derive(Debug)]
+pub struct StdIoError(E);
+
+#[cfg(not(feature = "no_std"))]
+impl StdIoError {
+    pub fn inner(&self) -> &E {
+        &self.0
+    }
+
+    pub fn into_inner(self) -> E {
+        self.0
+    }
+}
+
+#[cfg(not(feature = "no_std"))]
+impl PartialEq for StdIoError {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.kind() == other.0.kind()
+            && self.0.raw_os_error() == other.0.raw_os_error()
+            && self.0.to_string() == other.0.to_string()
+    }
+}
+
+#[cfg(not(feature = "no_std"))]
+impl Eq for StdIoError {}
+
+#[cfg(not(feature = "no_std"))]
+impl From<E> for StdIoError {
+    fn from(v: E) -> Self {
+        Self(v)
+    }
+}
+
 /// Converts the provided SV2 data type to a byte vector based on the SV2 encoding format.
 #[allow(clippy::wrong_self_convention)]
 pub fn to_bytes<T: Encodable + GetSize>(src: T) -> Result<Vec<u8>, Error> {
@@ -199,7 +233,7 @@ pub mod encodable {
 extern crate alloc;
 
 /// Error types used within the protocol library to indicate various failure conditions.
-#[derive(Debug, PartialEq, Eq, Clone)]
+#[derive(Debug, PartialEq, Eq)]
 pub enum Error {
     /// Indicates an attempt to read beyond a valid range.
     OutOfBound,
@@ -250,7 +284,7 @@ pub enum Error {
     #[cfg(not(feature = "no_std"))]
     /// Represents I/O-related errors, compatible with `no_std` mode where specific error types may
     /// vary.
-    IoError(E),
+    IoError(StdIoError),
 
     #[cfg(feature = "no_std")]
     /// Represents I/O-related errors, compatible with `no_std` mode.
@@ -289,7 +323,7 @@ impl From<E> for Error {
     fn from(v: E) -> Self {
         match v.kind() {
             ErrorKind::UnexpectedEof => Error::OutOfBound,
-            _ => Error::IoError(v),
+            _ => Error::IoError(v.into()),
         }
     }
 }

--- a/sv2/channels-sv2/Cargo.toml
+++ b/sv2/channels-sv2/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "channels_sv2"
-version = "6.0.0"
+version = "7.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -14,9 +14,9 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-binary_sv2 = { path = "../binary-sv2", version = "^5.0.0" }
-mining_sv2 = { path = "../subprotocols/mining", version = "^10.0.0" }
-template_distribution_sv2 = { path = "../subprotocols/template-distribution", version = "^5.1.0" }
+binary_sv2 = { path = "../binary-sv2", version = "^6.0.0" }
+mining_sv2 = { path = "../subprotocols/mining", version = "^11.0.0" }
+template_distribution_sv2 = { path = "../subprotocols/template-distribution", version = "^6.0.0" }
 tracing = { workspace = true }
 bitcoin = { workspace = true }
 primitive-types = { workspace = true }

--- a/sv2/codec-sv2/Cargo.toml
+++ b/sv2/codec-sv2/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "codec_sv2"
-version = "5.0.0"
+version = "6.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -12,9 +12,9 @@ homepage = "https://stratumprotocol.org"
 keywords = ["stratum", "mining", "bitcoin", "protocol"]
 
 [dependencies]
-framing_sv2 = { path = "../framing-sv2", version = "^6.0.0" }
+framing_sv2 = { path = "../framing-sv2", version = "^7.0.0" }
 noise_sv2 = { path = "../noise-sv2", default-features = false, optional = true, version = "^1.0.0" }
-binary_sv2 = { path = "../binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../binary-sv2", version = "^6.0.0" }
 buffer_sv2 = { path = "../buffer-sv2", version = "^3.0.0" }
 rand = { workspace = true, default-features = false }
 tracing = { workspace = true, optional = true }

--- a/sv2/extensions-sv2/Cargo.toml
+++ b/sv2/extensions-sv2/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "extensions_sv2"
-version = "0.1.0"
+version = "0.2.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -12,4 +12,4 @@ homepage = "https://stratumprotocol.org"
 keywords = ["stratum", "mining", "bitcoin", "protocol", "extensions"]
 
 [dependencies]
-binary_sv2 = { path = "../binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../binary-sv2", version = "^6.0.0" }

--- a/sv2/framing-sv2/Cargo.toml
+++ b/sv2/framing-sv2/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "framing_sv2"
-version = "6.0.1"
+version = "7.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -14,7 +14,7 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-binary_sv2 = { path = "../binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../binary-sv2", version = "^6.0.0" }
 buffer_sv2 = { path = "../buffer-sv2", optional=true, version = "^3.0.0" }
 noise_sv2 = { path = "../noise-sv2", version = "^1.0.0" }
 

--- a/sv2/handlers-sv2/Cargo.toml
+++ b/sv2/handlers-sv2/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "handlers_sv2"
-version = "0.4.0"
+version = "0.5.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -12,12 +12,12 @@ homepage = "https://stratumprotocol.org"
 keywords = ["stratum", "mining", "bitcoin", "protocol"]
 
 [dependencies]
-parsers_sv2 = { path = "../parsers-sv2", version = "^0.4.0"}
-binary_sv2 = { path = "../binary-sv2", version = "^5.0.0" }
-common_messages_sv2 = { path = "../subprotocols/common-messages", version = "^7.2.0" }
-framing_sv2 = { path = "../framing-sv2", version = "^6.0.0" }
-mining_sv2 = { path = "../subprotocols/mining", version = "^10.0.0" }
-template_distribution_sv2 = { path = "../subprotocols/template-distribution", version = "^5.1.0" }
-job_declaration_sv2 = { path = "../subprotocols/job-declaration", version = "^8.0.0" }
-extensions_sv2 = { path = "../extensions-sv2", version = "^0.1.0" }
+parsers_sv2 = { path = "../parsers-sv2", version = "^0.5.0"}
+binary_sv2 = { path = "../binary-sv2", version = "^6.0.0" }
+common_messages_sv2 = { path = "../subprotocols/common-messages", version = "^8.0.0" }
+framing_sv2 = { path = "../framing-sv2", version = "^7.0.0" }
+mining_sv2 = { path = "../subprotocols/mining", version = "^11.0.0" }
+template_distribution_sv2 = { path = "../subprotocols/template-distribution", version = "^6.0.0" }
+job_declaration_sv2 = { path = "../subprotocols/job-declaration", version = "^9.0.0" }
+extensions_sv2 = { path = "../extensions-sv2", version = "^0.2.0" }
 trait-variant = { workspace = true }

--- a/sv2/parsers-sv2/Cargo.toml
+++ b/sv2/parsers-sv2/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "parsers_sv2"
-version = "0.4.0"
+version = "0.5.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -12,13 +12,13 @@ homepage = "https://stratumprotocol.org"
 keywords = ["stratum", "mining", "bitcoin", "protocol"]
 
 [dependencies]
-binary_sv2 = { path = "../binary-sv2", version = "^5.0.0" }
-framing_sv2 = { path = "../framing-sv2", version = "^6.0.0" }
-common_messages_sv2 = { path = "../subprotocols/common-messages", version = "^7.2.0" }
-mining_sv2 = { path = "../subprotocols/mining", version = "^10.0.0" }
-template_distribution_sv2 = { path = "../subprotocols/template-distribution", version = "^5.1.0" }
-job_declaration_sv2 = { path = "../subprotocols/job-declaration", version = "^8.0.0" }
-extensions_sv2 = { path = "../extensions-sv2", version = "^0.1.0" }
+binary_sv2 = { path = "../binary-sv2", version = "^6.0.0" }
+framing_sv2 = { path = "../framing-sv2", version = "^7.0.0" }
+common_messages_sv2 = { path = "../subprotocols/common-messages", version = "^8.0.0" }
+mining_sv2 = { path = "../subprotocols/mining", version = "^11.0.0" }
+template_distribution_sv2 = { path = "../subprotocols/template-distribution", version = "^6.0.0" }
+job_declaration_sv2 = { path = "../subprotocols/job-declaration", version = "^9.0.0" }
+extensions_sv2 = { path = "../extensions-sv2", version = "^0.2.0" }
 
 [dev-dependencies]
 codec_sv2 = { path = "../codec-sv2", features = ["noise_sv2", "with_buffer_pool"] }

--- a/sv2/parsers-sv2/src/tlv/error.rs
+++ b/sv2/parsers-sv2/src/tlv/error.rs
@@ -1,7 +1,7 @@
 use core::fmt;
 
 /// Generic errors that can occur during TLV field encoding and decoding.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, PartialEq, Eq)]
 pub enum TlvError {
     /// TLV data buffer is too short to contain a valid TLV header or value.
     ///

--- a/sv2/subprotocols/common-messages/Cargo.toml
+++ b/sv2/subprotocols/common-messages/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "common_messages_sv2"
-version = "7.2.0"
+version = "8.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -14,7 +14,7 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-binary_sv2 = { path = "../../binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../../binary-sv2", version = "^6.0.0" }
 quickcheck = { workspace = true, optional = true }
 
 [features]

--- a/sv2/subprotocols/job-declaration/Cargo.toml
+++ b/sv2/subprotocols/job-declaration/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "job_declaration_sv2"
-version = "8.0.0"
+version = "9.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -13,4 +13,4 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 
 
 [dependencies]
-binary_sv2 = { path = "../../binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../../binary-sv2", version = "^6.0.0" }

--- a/sv2/subprotocols/mining/Cargo.toml
+++ b/sv2/subprotocols/mining/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "mining_sv2"
-version = "10.0.0"
+version = "11.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -15,7 +15,7 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-binary_sv2 = { path = "../../binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../../binary-sv2", version = "^6.0.0" }
 
 [dev-dependencies]
 quickcheck = { workspace = true }

--- a/sv2/subprotocols/template-distribution/Cargo.toml
+++ b/sv2/subprotocols/template-distribution/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "template_distribution_sv2"
-version = "5.1.0"
+version = "6.0.0"
 authors = ["The Stratum V2 Developers"]
 edition = "2021"
 readme = "README.md"
@@ -14,7 +14,7 @@ keywords = ["stratum", "mining", "bitcoin", "protocol"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-binary_sv2 = { path = "../../binary-sv2", version = "^5.0.0" }
+binary_sv2 = { path = "../../binary-sv2", version = "^6.0.0" }
 quickcheck = { workspace = true, optional = true }
 
 [features]

